### PR TITLE
Replace users completer with a topic's users completer

### PR DESCRIPTION
**Pull request by lezama**
Suggest just the users interacting in the current topic, works much better in larger installs where there are too many users.

Inspired in @dd32 https://github.com/dd32/wordpress.org/commit/7d3acbb72b22d9de20ec1d8db18f51581d5317fd

~TODO: The list of users should be added somewhere so this doesn't depend on the HTML markup.~

### CODE REVIEW

**Comment by dd32** [comment]
@lezama Thanks for bringing that upstream :)

Something I didn't add was if someone is @mention'd within a thread, but is not a thread participant, they won't appear in the mention completer (But they are in the existing WordPress.org completer, not the one this is inspired from).

> TODO: The list of users should be added somewhere so this doesn't depend on the HTML markup.

It would be helpful if this had a filter on it! That's something that would be beneficial for the W.org support forums.

It would also resolve paginated support threads where currently a participant from page 1 doesn't show as a completer on page 2 if they're not a poster on page2.. less of a real use-case, but it's come up before.

File: src/index.js [hunk 1]
```diff
@@ -161,6 +162,16 @@ domReady( () => {
 	// Modify any blocks we need to
 	addFilter( 'blocks.registerBlockType', 'blocks-everywhere/modify-blocks', modifyBlocks );
 
+	addFilter(
```
Comment (dd32) [inline src/index.js]: This seems like it needs a check to only start filtering if the bbPress support is enabled.

**Comment by johngodley** [comment]
This is nice and fast and seems a good solution. Using PHP to get the users across all topic pages seems a good improvement too. Now that it's in a build system we can also switch to using `import` instead of the `wp` globals, but not important right now.

You can use `wpBlocksEverywhere.settings.editorType` to determine if it's on a bbPress page. Ideally with the PHP based solution it could then work for the other places that the plugin supports, but at the moment that's low priority.

**Comment by lezama** [comment]
thanks for the quick 👀 

I am writing the list with PHP now! I think it's ready for more feedback :)

**Comment by johngodley** [comment]
Might need to make the list unique, otherwise works great.

image

**Comment by lezama** [comment]
ooops, lost the uniqueness at some point, should be good now.

**Comment by johngodley** [comment]
🚀

**Pull request merged**